WikiWatershed/model-my-watershed
# Parameterize DB parameter group

## Overview

When I tried to update the DataPlane stack in staging, CloudFormation encountered this error:

```
The following parameters are not defined for the specified group:
auto_explain.log_analyze, idle_in_transaction_session_timeout,
backend_flush_after, checkpoint_flush_after, auto_explain.sample_rate,
bgwriter_flush_after, auto_explain.log_format, auto_explain.log_timing,
force_parallel_mode, auto_explain.log_min_duration, auto_explain.log_triggers,
max_parallel_workers_per_gather, auto_explain.log_buffers,
gin_pending_list_limit, auto_explain.log_nested_statements,
auto_explain.log_verbose, log_replication_commands (Service: AmazonRDS; Status
Code: 400; Error Code: InvalidParameterValue; Request ID:
00280e40-fcac-427e-b913-e1850da4bd56)
```

This is because CloudFormation tries to update the parameter group family from `postgres9.4` to `postgres9.6`, rather than creating a new parameter group based off of `postgres9.6`. 

This PR parameterizes the DB parameter group name so we don't need to finesse CloudFormation.

Connects #3114 

### Checklist

- [x] `default.yml` is updated to reflect these changes

## Testing Instructions

Setup the `virtualenv` for `majorkirby`:

```bash
python2 -m virtualenv ./env
source ./env/bin/activate
pip install -r requirements.txt
```

Verify that valid JSON is returned for the DataPlane stack:

```bash
./mmw_stack.py launch-stacks \
    --aws-profile mmw-stg \
    --mmw-config-path default.yml \
    --mmw-profile staging \
    --data-plane \
    --print-json
```
